pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -421,21 +417,34 @@ func buildAutoMtlsSettings(
 		if len(subjectAltNamesToUse) == 0 {
 			subjectAltNamesToUse = serviceAccounts
 		}
-		return buildIstioMutualTLS(subjectAltNamesToUse, sniToUse, proxy), userSupplied
+		// For backward compatibility, use metadata certs if provided
+		if features.AllowMetadataCertsInMutualTLS && proxy.Metadata.TLSClientRootCert != "" {
+			return buildMutualTLS(subjectAltNamesToUse, sniToUse, proxy), autoDetected
```

Comment:
@howardjohn @ramaraochavali This is the core change, if metadata cert provided and AllowMetadataCertsInMutualTLS = true, we build the tls here. 

With this we can safely remove metadata cert stuff processing in `buildUpstreamClusterTLSContext`. All others are to fix the test related.

Reply:
Do we still need this flag?